src/events/components/CalendarView/CalendarEvent.tsx

```diff
@@ -0,0 +1,28 @@
+import Link from 'next/link';
+import React, { FC } from 'react';
+
+import HoverCard from 'common/components/HoverCard';
+import { getEventUrl } from 'core/appUrls';
+import { IEvent, getEventColor } from 'events/models/Event';
+
+import CalendarHoverCard from './CalendarHoverCard';
+
+import style from './CalendarEvent.less';
+
+interface IProps {
+  event: IEvent;
+}
+
+export const CalendarEvent: FC<IProps> = ({ event }) => {
+  return (
+    <Link {...getEventUrl(event.id)}>
+      <a>
+        <HoverCard card={<CalendarHoverCard eventId={event.id} />}>
```

Comment:
I feel like you've already explained this to me, but I don't remember the answer.
What is your thoughts about having the event here, passing the id to the CalendarHoverCard then getting the event from the store again from there?

Reply:
The Redux style guide has a good section on it: https://redux.js.org/style-guide/style-guide#connect-more-components-to-read-data-from-the-store

Reply:
Yes, I agree with the style guide! Their reasoning is that it can prevent rerendering of less DOM-elements by only caring about ID's until the object is actually used to prevent extra rendering.
However, don't we already have the `event` here? Thus, if anything the `event` changes, both `CalendarEvent` and `CalendarHoverCard` will rerender, since both have the `event` itself? If so, fetching it from the store again makes no difference. If the `CalendarTile` instead passed the `eventId` instead of the `event` to this component so that it were similar to the Redux Style Guide example, `CalendarEvent` would not rerender on event change, only the `CalendarHoverCard` would rerender. Seeing as we're using the `event` in both elements, only having the `id` would not work, so the two DOM-elements already are dependent on the same `event`-object and should rerender on the change anyway and can then pass the `event` itself. What do you think?

Reply:
I can agree there, and the current solution may not be the best. The initial idea was that the `CalendarHoverCard` was kind of a separate entity.

We can actually solve the re-rendering here in another way, by using minimal values until we actually need something else. We can select just the ids in `CalendarTile`, and select just the title and eventType in `CalendarEvent` to make it work with referential equality. Then pass the eventId to the card

Reply:
@FluidSense Had a go at what I mentioned above just now if you want to take a look :camera_flash:

Reply:
That should definitely deal with the rerendering, indeed!
A very good solution!